### PR TITLE
Feature /Tweaks for Cypress Framework and Add Jest Test Framework

### PR DESCRIPTION
**Pull request by PhotoNomad0**
# Describe what your pull request addresses

- cypress (https://github.com/unfoldingword/gateway-edit/issues/308)
  - add code coverage generation
  - add video capture of tests
  - support for other username/password in local testing of  01-app-init.spec.js
  - fix for race conditions in 01-app-init.spec.js
- jest (https://github.com/unfoldingword/gateway-edit/issues/309)
  - added jest test framework
  - initial unit tests

## Test Instructions

- [x] check that tests passed below under `Run Cypress and Jest Tests`

### CODE REVIEW

File: cypress/integration/01-app-init.spec.js [hunk 1]
```diff
@@ -7,17 +7,23 @@ describe('App login & initial setup', () => {
...
-    cy.get('[data-test="submit-button"]').click()
+    const USERNAME = Cypress.env('TEST_USERNAME')
+    const PASSWORD = Cypress.env('TEST_PASSWORD')
+
+    cy.get('input[name="username"]').should('be.visible').type(USERNAME)
+    cy.get('input[type="password"]').should('be.visible').type(PASSWORD)
 
-    cy.intercept('https://git.door43.org/api/v1/users/test_user01?noCache=**').as('getUser')
-    cy.intercept('https://git.door43.org/api/v1/users/test_user01/tokens?noCache=**').as('getToken')
+    // need to set up intercepts before final click, or sometimes had race condition that request was sent before had set up the intercept
```
Comment (mannycolon) [inline cypress/integration/01-app-init.spec.js:16]: Good job cleaning this up a bit.

File: package.json [hunk 2]
```diff
@@ -8,10 +8,15 @@
     "export": "next export",
     "postinstall": "husky install",
     "increment-build": "bash scripts/increment-build.sh && git add -A",
-    "cypress": "cypress open",
-    "cypress:headless": "cypress run",
-    "test": "start-server-and-test dev http://localhost:3000 cypress",
-    "test:headless": "start-server-and-test dev http://localhost:3000 cypress:headless"
+    "cypress": "NODE_ENV=test cypress open",
```
Comment (mannycolon) [inline package.json:11]: Great work adding these scripts! It will save us time in the future.

**Review by mannycolon — APPROVED** [review]
Great job overall

**Comment by mannycolon** [comment]
Pending local test

**Comment by mannycolon** [comment]
Seeing the following errors when running `yarn test:unit` locally.

Screen Shot 2021-11-09 at 11 23 43 AM
y

**Comment by PhotoNomad0** [comment]
> Seeing the following errors when running `yarn test:unit` locally. Is this okay?
> 

I noticed that also - took a look at the warning - seemed to be more of a warning than error.  Guessing that there are two configurations affecting the Cypress test.

**Pull request merged**